op-node/service.go

```diff
@@ -56,6 +56,9 @@ func NewConfig(ctx *cli.Context, log log.Logger) (*node.Config, error) {
 	}
 
 	l2SyncEndpoint := NewL2SyncEndpointConfig(ctx)
+	if l2SyncEndpoint.L2NodeAddr == "" { // optional config
+		l2SyncEndpoint = nil
```

Comment:
Changed it to handle it differently, and tested locally on goerli now; `Setup()` now returns nil if it's unset, and so the configuration is always present, but during init it can decide if the sync endpoint is set in the op node.